### PR TITLE
feat: add instructions to import in replit

### LINKED ISSUE #48274: Remove `.replit` and `replit.nix` Files From Boilerplate Repos
One of the Replit devs mentioned that they're making a lot of changes on their end, and recommends that we remove the `.replit` and `replit.nix` files from our boilerplate repos whenever possible. We should be able to do this in most of the boilerplate repos without issue.

There may be a couple of exceptions, though, like with the [Rust in Replit](https://github.com/freeCodeCamp/Rust-In-Replit) project boilerplate. For that project, there are some commands that need to run when the project is imported.

The recommendation for repos that need special commands is to leave the `.replit` and `replit.nix` files in the repo, but to make sure the only difference between them and the autogenerated `.replit` and `replit.nix` files are the commands that need to run.

Here's a list of the project boilerplates that should be updated. Please let me know if I missed any:

#### Practice Projects

- [x] Managing Packages with NPM
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-npm/
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-npm/pull/491
    - Learn description: 

- [x] Basic Node and Express
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-express/
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-express/pull/110
    - Learn description: 

- [x] MongoDB and Mongoose
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-mongomongoose/
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-mongomongoose/pull/51
    - Learn description: 

#### Required Projects

- [x] Timestamp Microservice
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-project-timestamp/
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-project-timestamp/pull/85
    - Learn description: 

- [x] Request Header Parser Microservice
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-project-headerparser/
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-project-headerparser/pull/31
    - Learn description: 

- [x] URL Shortener Microservice
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-project-urlshortener/
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-project-urlshortener/pull/53
    - Learn description: 

- [x] Exercise Tracker
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-project-exercisetracker/
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-project-exercisetracker/pull/67
    - Learn description: 

- [x] File Metadata Microservice
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-project-filemetadata/
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-project-filemetadata/pull/23
    - Learn description: 

#### Practice Projects

- [x] Quality Assurance and Testing with Chai
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-mochachai/
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-mochachai/pull/84
    - Learn description: 

- [x] Advanced Node and Express
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-advancednode/
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-advancednode/pull/26
    - Learn description: 

#### Required Projects

- [x] Metric-Imperial Converter
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-project-metricimpconverter/
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-project-metricimpconverter/pull/25
    - Learn description: 

- [x] Issue Tracker
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-project-issuetracker/
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-project-issuetracker/pull/26
    - Learn description: 

- [x] Personal Library
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-project-library
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-project-library/pull/18
    - Learn description: 

- [x] Sudoku Solver
  - Boilerplate repo: https://github.com/freecodecamp/boilerplate-project-sudoku-solver
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-project-sudoku-solver/pull/26
    - Learn description: 

- [x] American British Translator
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-project-american-british-english-translator/
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-project-american-british-english-translator/pull/22
    - Learn description: 

#### Required Projects

- [x] Arithmetic Formatter
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-arithmetic-formatter
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-arithmetic-formatter/pull/13
    - Learn description: 

- [x] Time Calculator
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-time-calculator
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-time-calculator/pull/7
    - Learn description: 

- [x] Budget App
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-budget-app
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-budget-app/pull/14
    - Learn description: 

- [x] Polygon Area Calculator
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-polygon-area-calculator
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-polygon-area-calculator/pull/7
    - Learn description: 

- [x] Probability Calculator
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-probability-calculator
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-probability-calculator/pull/10
    - Learn description: 

#### Required Projects

- [x] Mean-Variance-Standard Deviation Calculator
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-mean-variance-standard-deviation-calculator
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-mean-variance-standard-deviation-calculator/pull/8
    - Learn description: 

- [x] Demographic Data Analyzer
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-demographic-data-analyzer
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-demographic-data-analyzer/pull/12
    - Learn description: 

- [x] Medical Data Visualizer
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-medical-data-visualizer
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-medical-data-visualizer/pull/22
    - Learn description: 

- [x] Page View Time Series Visualizer
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-page-view-time-series-visualizer
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-page-view-time-series-visualizer/pull/11
    - Learn description: 

- [x] Sea Level Predictor
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-sea-level-predictor
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-sea-level-predictor/pull/9
    - Learn description: 

#### Practice Projects

- [x] Information Security with HelmetJS
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-infosec/
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-infosec/pull/54
    - Learn description: 

#### Required Projects

- [x] Stock Price Checker
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-project-stockchecker/
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-project-stockchecker/pull/30
    - Learn description: 

- [x] Anonymous Message Board
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-project-messageboard/
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-project-messageboard/pull/23
    - Learn description: 

- [x] Port Scanner
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-port-scanner
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-port-scanner/pull/5
    - Learn description: 

- [x] SHA-1 Password Cracker
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-SHA-1-password-cracker
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-SHA-1-password-cracker/pull/4
    - Learn description: 

- [x] Secure Real Time Multiplayer Game
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-project-secure-real-time-multiplayer-game/
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-project-secure-real-time-multiplayer-game/pull/20
    - Learn description: 

#### Required Projects

- [x] Rock Paper Scissors
  - Boilerplate repo: https://github.com/freeCodeCamp/boilerplate-rock-paper-scissors
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/boilerplate-rock-paper-scissors/pull/5
    - Learn description: 

## Special Cases

- [x] Rust in Replit
  - Tasks:
    - Import the project into Replit
    - Copy the autogenerated `.replit` and `replit.nix` files back into the boilerplate
    - Add the special commands back to the boilerplate's `.replit` file — the only diff between Replit's autogenerated `.replit` file and the one in the boilerplate are those special commands
  - Boilerplate repo: https://github.com/freeCodeCamp/Rust-In-Replit/
  - PRs:
    - Boilerplate: https://github.com/freeCodeCamp/Rust-In-Replit/pull/27
    - Learn description: N/A

### PR DESCRIPTION
**Pull request by Prince-Mendiratta**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Addresses #48274 

This PR targets adding the description to all the repos which have make use of a boilerplate to import the code on replit.

Most of the descriptions contain either of the following texts -
```
Curriculum:
- Use <a href="https://replit.com/github/freeCodeCamp/boilerplate-npm" target="_blank" rel="noopener noreferrer nofollow">our Replit starter project</a> to complete these challenges.
- Use <a href="https://replit.com/github/freeCodeCamp/boilerplate-project-headerparser" target="_blank" rel="noopener noreferrer nofollow">our Replit starter project</a> to complete your project.
- You will be <a href="https://replit.com/github/freeCodeCamp/boilerplate-demographic-data-analyzer" target="_blank" rel="noopener noreferrer nofollow">working on this project with our Replit starter code</a>.

Client:
Working on these challenges will involve you writing your code on Replit on our starter project. After completing each challenge you can copy your public Replit URL (to the homepage of your app) into the challenge screen to test it! Optionally you may choose to write your project on another platform but it must be publicly visible for our testing.
```
The instructions to import the project in replit have been added along these lines.

Signed-off-by: Prince Mendiratta prince.mendi@gmail.com

### CODE REVIEW

File: client/src/pages/learn/back-end-development-and-apis/basic-node-and-express/index.md
Comment (raisedadead) [inline client/src/pages/learn/back-end-development-and-apis/basic-node-and-express/index.md:18]:
```suggestion
Working on these challenges involves writing code on Replit in our starter project. Start by Importing the project on Replit. Next, you will see a <code>.replit</code> window. Select <code>Use run command</code> and click the <code>Done</code> button. Complete each challenge and copy the public Replit URL (to the homepage of your app) into the challenge screen to test it! Optionally, you may write your project on another platform, but it must be publicly visible for our testing.
```

**Review by raisedadead — COMMENTED** [review]
A quick comment, I think the tone should be more direct. But I am not going to hold up the PR. Here is a sample suggestion.

File: client/src/pages/learn/back-end-development-and-apis/basic-node-and-express/index.md
Comment (Prince-Mendiratta) [reply client/src/pages/learn/back-end-development-and-apis/basic-node-and-express/index.md:18]: This does indeed feel more descriptive, I'll make the required changes.

**Review by raisedadead — CHANGES_REQUESTED** [review]
OK - I think no one else has any issues with the verbiage; please go ahead with the changes I requested. Thanks a lot for working on this.

**Comment by Prince-Mendiratta** [comment]
Heya, @scissorsneedfoodtoo ! Any updates on this?

**Comment by Prince-Mendiratta** [comment]
Heya, @raisedadead! Any updates on this?

**Comment by scissorsneedfoodtoo** [comment]
Hi @Prince-Mendiratta, thank you for your patience, and thank you again for all of your hard work updating the text for these challenges. We really appreciate your help here.

I noticed a couple of small things. I'll leave some review comments, and hopefully it'll be easy to update everything with a quick find and replace.

File: client/src/pages/learn/back-end-development-and-apis/basic-node-and-express/index.md
Comment (scissorsneedfoodtoo) [inline client/src/pages/learn/back-end-development-and-apis/basic-node-and-express/index.md:20]: The word "importing" should be lowercase:

```suggestion
- Start by importing the project on Replit.
```

File: curriculum/challenges/english/05-back-end-development-and-apis/back-end-development-and-apis-projects/exercise-tracker.md
Comment (scissorsneedfoodtoo) [inline curriculum/challenges/english/05-back-end-development-and-apis/back-end-development-and-apis-projects/exercise-tracker.md:17]: Thanks for customizing the text for these projects. It's great that you refer to Replit as an option, then go into detail if learners go that route.

But "setup" should be in the two word verb form here. We could also simplify things a bit:

```suggestion
If you use Replit, follow these steps to set up the project:
```

**Review by scissorsneedfoodtoo — CHANGES_REQUESTED** [review]

File: client/src/pages/learn/back-end-development-and-apis/basic-node-and-express/index.md
Comment (Prince-Mendiratta) [reply client/src/pages/learn/back-end-development-and-apis/basic-node-and-express/index.md]: I totally missed on that haha :P

**Comment by scissorsneedfoodtoo** [comment]
Awesome. Thanks for making those changes so quickly.

I'm doing a final pass now to make sure all the steps you added show up on all the necessary project pages, or on the first step for any of the practice projects.

**Comment by scissorsneedfoodtoo** [comment]
Just finished going over everything, and it seem like only the Rock Paper Scissors project is missing these steps: https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/11-machine-learning-with-python/machine-learning-with-python-projects/rock-paper-scissors.md

Would you be able to add the steps to that file as well? Once that's updated, this should be good to go unless anyone else has other suggestions.

**Review by scissorsneedfoodtoo — APPROVED** [review]
Awesome. Thanks again for all of your help updating all of these challenges, @Prince-Mendiratta.

Everything LGTM 👍

**Review by raisedadead — APPROVED** [review]
LGTM. Thanks for your hard work @Prince-Mendiratta

**Pull request merged**
